agent/src/main/java/io/type/pollution/agent/TraceInstanceOf.java

```diff
@@ -101,6 +101,13 @@ public void onTypeCheckHit(Class interfaceClazz, String trace) {
             LAST_SEEN_INTERFACE_UPDATER.lazySet(this, interfaceClazz);
             if (lastSeen != null) {
                 updateTraceCount(interfaceClazz, trace);
+
+                TypeCheckThrashEvent event = new TypeCheckThrashEvent();
```

Comment:
https://github.com/RedHatPerf/type-pollution-agent/blob/53241201ad2adf317fe10e1caba083cb63ae73dd/agent/src/main/java/io/type/pollution/agent/TraceInstanceOf.java#L81 can you add a Miss one too - or use some field to distinguish between them?

IDK if you plan to have it running in a proper benchmark but this can be a VERY high frequency event, and I've designed it to run in the hot path, at the time e.g. traces are pre-computed on bytecode transformation
if you plan to do something similar, maybe you just want to emit the summary event instead - like the report - on demand or at intervals or using a similar trick than the new TLAB JFR events (by using a counter which "overflow" and emit events only if it overflows).
If is not meant to be used in a real benchmark (but sadly that's the easier way to make it happen too :( ) let's leave it as it is and in case, change it in something cheaper after, if it create problems.

Reply:
> I am not using this inside a benchmark

yeah I can see it why but remember that the "pollution" pattern depends by both the ordering of type checks - which can trash the cache using a single thread OR because of high level of concurrency - which means you need an option to do it in a proper real world benchmark too, although it will be used just to detect that there's some trashing going on and fail a release - for example. Something which cannot be done with simple unit tests, really.

What can be done here, to enable using this in high-throughput scenario is:
i.e.

1. use a atomic reference field updater which contains the current `TypeCheckThrashEvent` stored within the counter instance: it has to contain an atomic long with the number of accumulated thrashed events
2. when the current one overflow - we can perform a compareAndSet(usedEvent, null) pro-bono: if it fails is not important
3. any racy thread which overflow > 1 more past the threshold of the current event, can try create a new initialized event with compareAndSet(usedEvent, new Event(1)) - if it failes, can retry or give up, really - but with compareAndExchange, in theory can already check if there's a newer event ready to pick it

In a jfr unit test, the default threshold can be 1: we don't care about performance and should "just work": in a benchmark we can have 1000 or 10000 and the actual event count must be scaled using the limit, as per TLAB allocation events, in JFR.

Let me know if it seems appealing for you: if not we can go through this one, but when you'll need it for benchmark, we should get back to this, eventually.